### PR TITLE
`cmd` and `endpoint` should expand `$(location)`

### PR DESCRIPTION
**Pull request by sfc-gh-ptabor**
When developing some macros on top of OCI rules, I discovered that if `cmd` (and `endpoint`) for consistency was able to resolve path to the target file (frequently transitively embedded within a tarball), than the macros could not require from user taking care of proper full paths of artifacts. Please see the attached example and go doc update.

### CODE REVIEW

File: oci/private/deps_aspect.bzl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
+  An aspect that find's most transitive dependencies (target) of the given rule's attribute.
```
Comment (seh) [inline oci/private/deps_aspect.bzl:2]: Should "target" be pluralized? It's not clear whether you mean that dependencies are targets, vice versa, or neither.
```suggestion
  An aspect that finds most transitive dependencies (target) of the given rule's attribute.
```

Comment (sfc-gh-ptabor) [reply oci/private/deps_aspect.bzl]: Corrected.

**Comment by thesayyn** [comment]
Hey @sfc-gh-ptabor.  Thanks for taking action and creating the PR.  I have a few concerns about the necessity of this but would like to write them into a discussion issue. I'd appreciate it if you could create an issue so we can have a discussion first involving other parties.

**Comment by sfc-gh-ptabor** [comment]
@thesayyn: Thank you for taking the look. The issue is: https://github.com/bazel-contrib/rules_oci/issues/287

**Comment by thesayyn** [comment]
> @thesayyn: Thank you for taking the look. The issue is: #287

Thanks! Ideally we should just make `entrypoint` and `cmd` stampable. See: https://github.com/bazel-contrib/rules_oci/issues/287#issuecomment-1633036519

**Comment by sfc-gh-ptabor** [comment]
Closing and it got suppressed by https://github.com/bazel-contrib/rules_oci/pull/297

**Pull request closed without merging**